build_projects/shared-build-targets-utils/Utils/Dirs.cs

```diff
@@ -47,9 +47,9 @@ private static string GetNuGetPackagesDir()
         {
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
-                return Path.Combine(Environment.GetEnvironmentVariable("USERPROFILE"), ".nuget", "packages");
+                return Path.Combine(Dirs.RepoRoot, ".nuget", "packages");
```

Comment:
Check this: https://github.com/dotnet/cli/blob/rel/1.0.0/src/Microsoft.DotNet.ProjectModel/Resolution/PackageDependencyProvider.cs#L159.

You don't necessarily need to use Dirs.RepoRoot. You can look at the env variable here as well.